app/src/main/java/org/kiwix/kiwixmobile/main/KiwixNewNavigationActivity.kt

```diff
@@ -0,0 +1,92 @@
+/*
+ * Kiwix Android
+ * Copyright (c) 2020 Kiwix <android.kiwix.org>
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package org.kiwix.kiwixmobile.main
+
+import android.content.Intent
+import android.os.Bundle
+import android.view.Menu
+import androidx.appcompat.view.ActionMode
+import androidx.navigation.NavController
+import androidx.navigation.findNavController
+import androidx.navigation.ui.AppBarConfiguration
+import androidx.navigation.ui.setupWithNavController
+import com.google.android.material.bottomnavigation.BottomNavigationView
+import org.kiwix.kiwixmobile.R
+import org.kiwix.kiwixmobile.core.base.BaseFragmentActivityExtensions
+import org.kiwix.kiwixmobile.core.di.components.CoreComponent
+import org.kiwix.kiwixmobile.core.main.CoreMainActivity
+import org.kiwix.kiwixmobile.kiwixActivityComponent
+
+class KiwixNewNavigationActivity : CoreMainActivity() {
+  private lateinit var navController: NavController
+  private lateinit var appBarConfiguration: AppBarConfiguration
+  private var actionMode: ActionMode? = null
+
+  override fun injection(coreComponent: CoreComponent) {
+    kiwixActivityComponent.inject(this)
+  }
+
+  private val finishActionModeOnDestinationChange =
+    NavController.OnDestinationChangedListener { controller, destination, arguments ->
+      actionMode?.finish()
+    }
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_new_navigation)
+    val navView: BottomNavigationView = findViewById(R.id.nav_view)
```

Comment:
shouldn't need findViewById unless this is across the module boundary which it probably shouldn't be